File: Front End Angular/src/app/services/biglietto.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {finalize, Observable} from 'rxjs';
import {InsertBigliettoDTO} from '../dto/biglietto/request/insert-biglietto-dto';
import {ResponseBigliettoDTO} from '../dto/biglietto/response/response-biglietto-dto';
import {SharedService} from "./shared.service";

@Injectable({
  providedIn: 'root'
})
export class BigliettoService {
  private readonly BASE_URL = 'http://localhost:8080';

  constructor(private http: HttpClient) {}


  insert(dto: InsertBigliettoDTO): Observable<ResponseBigliettoDTO[]> {
    return this.http.post<ResponseBigliettoDTO[]>(
      `${this.BASE_URL}/cliente/biglietto`,
      dto
    );
  }


  findByIdUtente(idUtente: number): Observable<ResponseBigliettoDTO[]> {

    return this.http.get<ResponseBigliettoDTO[]>(
      `${this.BASE_URL}/staff/biglietto/utente/${idUtente}`
    );
  }


  findByIdSpettacolo(idSpettacolo: number): Observable<ResponseBigliettoDTO[]> {

    return this.http.get<ResponseBigliettoDTO[]>(
      `${this.BASE_URL}/staff/biglietto/spettacolo/${idSpettacolo}`
    );
  }


  removeById(id: number, email: string): Observable<void> {
    return this.http.delete<void>(`${this.BASE_URL}/cliente/biglietto/${id}`, {
      body: email,
    });
  }



  clientebiglietti(): Observable<ResponseBigliettoDTO[]> {
    return this.http.get<ResponseBigliettoDTO[]>(
        `${this.BASE_URL}/cliente/biglietto`
    );
  }

}

File: Back End Spring/src/main/java/org/elis/movieexplorer/service/definition/BigliettoService.java
package org.elis.movieexplorer.service.definition;

import java.util.List;

import org.elis.movieexplorer.dto.biglietto.request.InsertBigliettoDTO;
import org.elis.movieexplorer.dto.biglietto.response.ResponseBigliettoDTO;
import org.elis.movieexplorer.model.Biglietto;
import org.elis.movieexplorer.model.Utente;

public interface BigliettoService {
	// CREATE
	List<ResponseBigliettoDTO> insert(InsertBigliettoDTO dto, Utente utenteLoggato);
	
	// READ
	List<ResponseBigliettoDTO> findByIdUtente(Long idUtente);
	
	List<ResponseBigliettoDTO> findByIdSpettacolo(Long idSpettacolo);
	
	// UPDATE
	//----
	
	// DELETE
	void removeByIdEmail(Long idBiglietto, String email);
}